src/private/main.rs

```diff
@@ -259,6 +259,38 @@ fn process_request(
         return Ok(mirror_response);
     }
 
+    // GetBlockInfoRequest
+    if query_request.has_get_block_info() {
+        let mirror_request = query_request.get_get_block();
```

Comment:
There's `get_block` and `get_block_info`. You're checking for `has_get_block_info` and then calling `get_get_block`...

Reply:
Fixed in #7